nav2_route/src/goal_intent_extractor.cpp

```diff
@@ -53,6 +58,49 @@ void GoalIntentExtractor::configure(
   nav2_util::declare_parameter_if_not_declared(
     node, "min_dist_from_start", rclcpp::ParameterValue(0.10));
   min_dist_from_start_ = static_cast<float>(node->get_parameter("min_dist_from_start").as_double());
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, "num_of_nearest_nodes", rclcpp::ParameterValue(3));
+  int num_of_nearest_nodes = node->get_parameter("num_of_nearest_nodes").as_int();
+
+  node_spatial_tree_ = std::make_shared<NodeSpatialTree>(num_of_nearest_nodes);
+  node_spatial_tree_->computeTree(graph);
+
+  nav2_util::declare_parameter_if_not_declared(
+    node, "enable_search", rclcpp::ParameterValue(false));
+  enable_search_ = node->get_parameter("enable_search").as_bool();
+
+  if (enable_search_) {
+    std::string global_costmap_topic;
+    nav2_util::declare_parameter_if_not_declared(
+      node, "global_costmap_topic",
+      rclcpp::ParameterValue(std::string("global_costmap/costmap_raw")));
+    node->get_parameter("global_costmap_topic", global_costmap_topic);
+
+    nav2_util::declare_parameter_if_not_declared(
+      node, "max_iterations", rclcpp::ParameterValue(500));
+    max_iterations_ = node->get_parameter("max_iterations").as_int();
+
+    nav2_util::declare_parameter_if_not_declared(
+      node, "enable_search_viz", rclcpp::ParameterValue(false));
+    enable_search_viz_ = node->get_parameter("enable_search_viz").as_bool();
+
+    costmap_sub_ =
+      std::make_unique<nav2_costmap_2d::CostmapSubscriber>(node, global_costmap_topic);
+    bfs_ = std::make_unique<BreadthFirstSearch>();
+    start_expansion_viz_ = node->create_publisher<nav_msgs::msg::OccupancyGrid>(
+      "start_expansions",
+      1);
+    goal_expansion_viz_ =
+      node->create_publisher<nav_msgs::msg::OccupancyGrid>("goal_expansions", 1);
+  }
+}
+
+void GoalIntentExtractor::activate()
+{
+  if (enable_search_) {
+    bfs_->initialize(costmap_sub_->getCostmap().get(), max_iterations_);
```

Comment:
getCostmap() returns a shared pointer, that's means for exactly this purpose silly :-)